### PR TITLE
Birthdate Input Form

### PR DESCRIPTION
**Pull request by psollars**
Adds separate form fields for birthdate and converts the input, May 6 1990, into an ISO string '1990-05-06' (slicing off the Time)
If the month is misspelled or entered as a number an alert is triggered, maybe a dropdown or autocomplete is a better way to cut down input errors?

### CODE REVIEW

**Comment by psollars** [comment]
Adding parseInt to date constructor cleaned up this error...
[at-loader] ./web_client/app/registration-check/registration-check.tsx:49:32
TS2345: Argument of type 'string' is not assignable to parameter of type 'number'.

**Comment by slothwerks-studio** [comment]
Might we go with a numeric month?  This would be language neutral and might be easier to control input-wise.  With the current application layout, there is no indication that the month should be spelled out, resulting in confusion when I enter "01" as a month and the error box says "enter a valid month".  Perhaps placeholder text of MM DD YYYY?

**Comment by diiq** [comment]
@slothwerks-studio the goal is to move to a dropdown for month, rather than a numeric input.

The question is just whether to use a dropdown that can be custom-styled, or just use the browser default.

**Comment by slothwerks-studio** [comment]
@diiq Oh!  My apologies; I was unaware that was the question at this point.  Given the existing styles on the app, I would recommend implementing our own CSS in this case.  Does the team need someone to tackle this task?

**Comment by psollars** [comment]
@slothwerks-studio I messed with an autocomplete component and then put that on a new branch 'month-autocomplete', some of the TypeScript stuff is tripping me up so this is definitely not functional. It's probably best to create a very simple dropdown on this current branch first.

**Comment by jacebrowning** [comment]
@psollars Are you still interested in working on this?

**Comment by jacebrowning** [comment]
@diiq What is the value of "... implicitly has an 'any' type." for this project? Is there some way to resolve that automatically?

**Comment by diiq** [comment]
@jacebrowning We should explicitly type all variables; typing means that the next dev doesn't need to do nearly as much head-scratching investigation, their editor will offer sensible autocomplete suggestions, etc.

I'll take a look.

File: web_client/app/models/voter/voter.tsx [hunk 1]
```diff
@@ -13,7 +13,7 @@ export class Voter {
 
   @action
   checkRegistration() {
-    return VoterService.checkRegistration(this.firstName, this.lastName, this.birthDate, this.zipCode).then(
+    return VoterService.checkRegistration(this.firstName, this.lastName, this.birthDate.toISOString().slice(0, 10), this.zipCode).then(
```
Comment (diiq) [inline web_client/app/models/voter/voter.tsx]: Need to make this change in `signUp`, too.

**Comment by jacebrowning** [comment]
> can we get the API to take any ISO-formatted date?

FYI: The reason this is happening is because the the API is expecting a date, not a date-time.

**Comment by diiq** [comment]
@psollars This looked really good! I moved the birthdate stuff into its own component to keep the registration form a little cleaner, and I tidied up here and there -- but it looks really great. 

I'm pushing the button -- let's get it live!

**Pull request merged**